js/guided-tour.js

```diff
@@ -2,83 +2,69 @@ import { Application, Controller } from "/js/stimulus.js"
-    viewer.viewport.goHome();
-  }
+    if (window.location.hash && this.slidesTargets.findIndex(x => x.id == window.location.hash.substring(1)) > 0) {
+      this.index = this.slidesTargets.findIndex(x => x.id == window.location.hash.substring(1));
+    }
 
-  reset() {
-    this.index = -1;
-    this.intialScreenTarget.classList.remove('d-none');
-    this.slidesTarget.classList.add('d-none');
-    this.endScreenTarget.classList.add('d-none');
+    this.render();
   }
 
-  end() {
-    this.initialScreenTarget.classList.add('d-none');
-    this.slidesTarget.classList.add('d-none');
-    this.endScreenTarget.classList.remove('d-none');
-    this.captionTarget.classList.add('d-none');
-    viewer.viewport.goHome();
+  // reset the slideshow back to the intro card
+  reset() {
+    this.setIndex(0);
   }
 
+  // paginate to the next slide, or the end card
   next() {
-    if ((this.index + 1) >= Object.entries(this.json).length) {
-      return this.end();
-    }
-
-    this.index++;
-    this.showCurrentItem();
+    this.setIndex(this.index + 1);
   }
 
+  // paginate to the previous slide, or the intro card
   previous() {
-    if (this.index <= 0) {
-      return this.reset();
-    }
+    this.setIndex(this.index - 1);
+  }
 
-    this.index--;
-    this.showCurrentItem();
+  setIndex(index) {
+    this.index = Math.max(0, Math.min(index, this.slidesTargets.length));
+    this.render();
   }
 
+  // @private
+  // get the currently displayed item
   getItem() {
-    const [key, value] = Object.entries(this.json)[this.index];
-
-    return value;
+    return this.slidesTargets[this.index];
   }
 
-  showCurrentItem() {
+  render() {
     const item = this.getItem(this.index);
 
-    if (item.viewport) {
-      viewer.viewport.panTo(viewer.viewport.imageToViewportCoordinates(item.viewport.x, item.viewport.y));
-      viewer.viewport.zoomTo(item.viewport.zoom);
-    }
+    this.slidesTargets.forEach(x => x.classList.add('d-none'));
+    item.classList.remove('d-none');
 
-    if (item.bio) {
-      this.bioTarget.classList.remove('d-none');
-      const card = document.createElement('div');
-      const title = document.createElement('h4');
-      title.innerHTML = item.bio.title;
-      card.prepend(title);
-      this.bioTarget.replaceChildren(card);
-    } else {
-      this.bioTarget.classList.add('d-none');
+    this.slideContainerTargets.forEach(container => {
+      if (container.contains(item)) {
+        container.classList.remove('d-none');
+      } else {
+        container.classList.add('d-none');
+      }
+    });
+
+    const data = JSON.parse(item.querySelector("script[type='application/json']")?.innerHTML || null) ?? {};
+
+    if (data.viewport) {
+      viewer.viewport.panTo(viewer.viewport.imageToViewportCoordinates(data.viewport.x, data.viewport.y));
+      viewer.viewport.zoomTo(data.viewport.zoom);
+    } else if (this.index == 0 || (this.index + 1) == this.slidesTargets.length) {
+      viewer.viewport.goHome();
```

Comment:
initially saw this and was confused about why `viewer` is defined in this scope. doing `window.viewer` here and in the medical devices `index.html` might help make it a little more explicit, although obviously the result is the same. alternatively we could do something like `this.viewer = window.viewer` in `connect()` (and maybe even error with a nicer message if the viewer isn't available)